modules/Microsoft.Compute/galleries/application/deploy.bicep

```diff
@@ -0,0 +1,100 @@
+@description('Required. Name of the application definition.')
+param name string
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Conditional. The name of the parent Azure Compute Gallery. Required if the template is used in a standalone deployment.')
+@minLength(1)
+param galleryName string
+
+@description('Optional. The description of this gallery Application Definition resource. This property is updatable.')
+param applicationDefinitionDescription string = ''
```

Comment:
@shawntmeyer very good job, again! Just 2 main points are missing for a child module PR to be complete:

1. Whenever a new child module is added, we aim to add the option to deploy one or more occurrences of that child resource type in the parent module, i.e., in this case the gallery module should allow to deploy one or multiple gallery applications, same as it allows to deploy multiple gallery images.
2. Next, to validate the new child module properties, the parent common test should be extended to also include the deployment of at least one occurrence of the child, i.e., in this case, the gallery common test should include the deployment of one gallery application, same as it includes the deployment of one gallery image.

Reply:
I added the parent common test and updated the parent (Galleries) deploy.bicep. Updated API version as well.